### PR TITLE
Option to disable index array lengths

### PR DESCRIPTION
**Pull request by brkolla**
@davisp . I have started fresh again as I had to make additional changes so that the mango index service can play nice with the new field added to the index doc.
Have also added two new tests.
Please review these changes again and will deliver once I get a +1 from you. I will also make these changes in dbnext too. FYI @tonysun83

### CODE REVIEW

**Comment by davisp** [comment]
LGTM but you should just update the existing PR rather than creating a new one.

**Comment by brkolla** [comment]
I messed up the old PR, and I had delivered/merged that PR with a typo (happened while incorporating the style changes) and as a result I had to revert that.

**Comment by brkolla** [comment]
@davisp, based on my explanation above on why I used the new PR and couldn't use the old one, can you please +1 this PR so that I can merge.

**Comment by davisp** [comment]
I don't understand what happened to the other PR that you can put these commits on that branch.

**Comment by brkolla** [comment]
Here is what happened, I basically had a typo while incorporating the last set of coding style changes, specifically the change about the record definition to add new field and had an extra "r" in array_length field.
And I realized that typo mistake after merging the PR, and I had to revert that merge. And I thought once you merge the PR and revert it you can't use the same PR. And that's I ended up with this new PR.

**Comment by brkolla** [comment]
@davisp ? Please let me know if it's ok to merge this. I know these changes are +1 ed on the old PR and I don't want to deliver until I get +1 on this one too and also want to make sure creating the new PR won't cause any issues. I will have to handle this better if ever had to revert again.

File: test/mango.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def create_index(self, fields, idx_type="json", name=None, ddoc=None):
         return r.json()["result"] == "created"
 
     def create_text_index(self, analyzer=None, selector=None, idx_type="text",
-        default_field=None, fields=None, name=None, ddoc=None):
+        default_field=None, fields=None, name=None, ddoc=None,index_array_lengths=None):
```
Comment (tonysun83) [inline test/mango.py]: need a space here

Comment (brkolla) [inline test/mango.py]: I did pep8 check and it didn't complain about it

**Comment by rnewson** [comment]
+1

**Comment by tonysun83** [comment]
+1 after documenting the $size issue.

File: test/mango.py (hunk 1 above)
Comment (robfraz) [inline test/mango.py]: Not sure what pep8 you're running @brkolla ...
```test/mango.py:105:9: E125 continuation line with same indent as next logical line
test/mango.py:105:9: E128 continuation line under-indented for visual indent
test/mango.py:105:62: E231 missing whitespace after ','
test/mango.py:105:80: E501 line too long (88 > 79 characters)```

**Pull request merged**